cpp/include/cudf/utilities/span.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+#include <type_traits>
+
+namespace cudf {
+namespace detail {
+
+constexpr size_t dynamic_extent = std::numeric_limits<std::size_t>::max();
+
+/**
+ * @brief C++20 std::span with reduced feature set.
+ */
+template <typename T, size_t Extent, typename Derived>
+class span_base {
+  static_assert(Extent == dynamic_extent, "Only dynamic extent is supported");
+
+ public:
+  using element_type    = T;
+  using value_type      = std::remove_cv<T>;
+  using size_type       = std::size_t;
+  using difference_type = std::ptrdiff_t;
+  using pointer         = T*;
+  using iterator        = T*;
+  using const_pointer   = T const*;
+  using reference       = T&;
+  using const_reference = T const&;
+
```

Comment:
Missing the `extent` member. 
```suggestion
  static constexpr std::size_t extent = Extent;
```